src/lib.rs

```diff
@@ -6,28 +6,30 @@ use lrpar::{lrpar_mod, LexParseError, NonStreamingLexer};
 use scope::Scope;
 use std::{cell::RefCell, collections::HashMap, rc::Rc};
 
-lrlex_mod!("calc.l");
-lrpar_mod!("calc.y");
+lrlex_mod!("yaiwr.l");
+lrpar_mod!("yaiwr.y");
 
 pub mod ast;
 pub mod bytecode;
 pub mod err;
+pub mod hash;
 pub mod instruction;
 pub mod scope;
 
 use ast::AstNode;
 use err::InterpError;
+use hash::HashId;
 
 use crate::instruction::JumpInstruction;
 
-pub struct Calc {
-    fun_store: HashMap<String, Instruction>,
+pub struct YIWR {
+    fun_store: HashMap<u64, Instruction>,
```

Comment:
I thought about moving function declarations to the Scope as well, but it was complicated, so I left it here for now.